network/common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -158,6 +158,13 @@ public TransportClient createClient(String remoteHost, int remotePort) throws IO
     }
   }
 
+  /**
+   * Create a completely new {@link TransportClient} to the given remote host / port */
+  public TransportClient createNewClient(String remoteHost, int remotePort) throws IOException {
```

Comment:
I actually wanted to add this method for a different change (no PR yet), so it's welcome. But the javadoc needs to mention that this connection is not pooled, like the ones returned by `createClient`.

Because of that, I'd also name the method `createUnmanagedClient`, and make it clear that it's the caller's responsibility to close it.